speechbrain/core.py

```diff
@@ -1200,11 +1211,18 @@ def _wrap_distributed(self):
             for name, module in self.modules.items():
                 if any(p.requires_grad for p in module.parameters()):
                     module = SyncBatchNorm.convert_sync_batchnorm(module)
-                    module = DDP(
-                        module,
-                        device_ids=[self.device],
-                        find_unused_parameters=self.find_unused_parameters,
-                    )
+                    if self.distributed_backend == "gloo":
```

Comment:
Why has gloo a different behavior now?

Reply:
I think @anautsch might have assumed that `gloo` is only desirable/used in a CPU DDP scenario. [This is the general recommendation,](https://pytorch.org/docs/stable/distributed.html#which-backend-to-use) but using `gloo` for GPU-based DDP appears to be a supported option, even if the recommendation is `nccl`.

Reply:
`gloo` is the only way for CPU to do DDP - i.e. what it wants, must be there for testing when GPUs are not there (github workflows).

`gloo` has API differences to the other two, as it uses here one CPU node only, and not several GPU nodes. Otherwise, pytorch throws an error - and that's it then.

Reply:
btw - I did not try to make `mpi` available, compare
https://pytorch.org/docs/stable/distributed.html

Reply:
`gloo` cannot do GPU !!!

Reply:
a DDP integration testing -> not possible w/o GPUs
no idea how to make that workable within github workflows ;)
could be done offline & before another release is published - but then, what should the test case look like?